code/main/media/mediaModule.brs

```diff
@@ -41,140 +41,47 @@ function _adb_MediaModule(configurationModule as object, edgeRequestQueue as obj
         ' Event data example:
         ' {
         '     clientSessionId: "xx-xxxx-xxxx",
-        '     tsObject: { ts_inISO8601: "2019-01-01T00:00:00.000Z", ts_inMillis: 1546300800000 },
+        '     tsObject: { tsInISO8601: "2019-01-01T00:00:00.000Z", tsInMillis: 1546300800000 },
         '     xdmData: { xdm: {} }
         '     configuration: { }
         ' }
         processEvent: sub(requestId as string, eventData as object)
-            mediaEventType = eventData.xdmData.xdm.eventType
-            clientSessionId = eventData.clientSessionId
-            tsObject = eventData.tsObject
-            xdmData = eventData.xdmData
+            eventType = eventData.xdmData.xdm.eventType
 
-            if mediaEventType = m._CONSTANTS.MEDIA.SESSION_START_EVENT_TYPE
-                sessionConfig = eventData.configuration
-                m._startSession(clientSessionId, sessionConfig, xdmData, tsObject)
+            if not m._isValidMediaEvent(eventType) then
+                _adb_logError("processEvent() - Cannot process media event (" + FormatJson(eventType) + "), media event type (" + FormatJson(eventType) + ") is invalid.")
                 return
             end if
 
-            if mediaEventType <> invalid and _adb_isStringInArray(mediaEventType, m._CONSTANTS.MEDIA.EVENT_TYPES)
-                m._trackEventForSession(requestId, clientSessionId, xdmData, tsObject)
-            else
-                _adb_logError("processEvent() - media event type is invalid: " + FormatJson(mediaEventType))
-            end if
-        end sub,
-
-        _isMediaConfigReady: function() as boolean
-            ' appVersion is optional for sessionStart request
-            if _adb_isEmptyOrInvalidString(m._configurationModule.getMediaChannel()) or _adb_isEmptyOrInvalidString(m._configurationModule.getMediaPlayerName())
-                return false
-            end if
-            return true
-        end function,
-
-        _startSession: sub(clientSessionId as string, sessionConfig as object, xdmData as object, tsObject as object)
-            ' TODO: validate and sanitize the sessionConfig
-            ' TODO: the session-level config should be merged with the global config, before the validation
-
-            if not m._isMediaConfigReady() then
-                _adb_logError("_startSession() - the media session is not created/started properly (missing the channel name or the player name).")
-                return
-            end if
-
-            m._sessionManager.createNewSession(clientSessionId)
-            ' TODO: add session-level config to the sessionManager
-            ' m._sessionManager.createNewSession(sessionConfig,sessionConfig["config.mainpinginterval"],sessionConfig["config.adpinginterval"])
+            sessionConfig = eventData.configuration
+            clientSessionId = eventData.clientSessionId
+            tsObject = eventData.tsObject
+            xdmData = eventData.xdmData
```

Comment:
nit: Create mediaHit here and pass it everywhere.

Reply:
Yes, I was thinking about this. Let me move createMediaHit to mediaModule.